File: TestingSystem/API/Tests.Api/Tests.Api/Models/TakeQuestionModel.cs
namespace Tests.Api.Models;

public class TakeQuestionModel
{
    public string? QuestionId { get; set; }
    public string? QuestionName { get; set; }
    public string? QuestionType { get; set; }
    public IEnumerable<TakeSelectableAnswerModel>? SelectableQuestions { get; set; }
}

File: TestingSystem/Frontend/src/app/core/models/take-question-model.ts
import {TakeSelectableAnswerModel} from "./take-selectable-answer-model";

export class TakeQuestionModel {
  constructor(
    public questionId?: string,
    public questionName?: string,
    public questionType?: string,
    public selectableQuestions?: TakeSelectableAnswerModel[]
  ) {
  }
}
